pkg/rbac/authenticationMiddleware_test.go

```diff
@@ -1,2 +1,137 @@
 // Copyright Contributors to the Open Cluster Management project
 package rbac
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+//test valid token from cookie
+func TestTokenCookieAuthenticated(t *testing.T) {
+
+	token := "validtoken"
+	authenticateHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+
+	r := httptest.NewRequest("GET", "https://localhost:4010/playground", nil)
```

Comment:
The actual API should be at `POST https://localhost:4010/searchapi/graphql`.  It doesn't make a difference here because we are just mocking, but could you update for correctness.

```suggestion
	r := httptest.NewRequest("POST", "https://localhost:4010/searchapi/graphql", nil)
```